src/lib/modules/query/parser.js

```diff
@@ -1,10 +1,10 @@
 'use strict';
 
-const VALID_QUERY_REGEX = new RegExp(/db\.(a-zA-Z1-9)*.*/);
-const COLLECTION_FROM_QUERY_REGEX = /(?:db\.)([a-zA-Z1-9]*)[.]{0,1}/; //extracts the collection name from a valid query
-const FUNCTION_FROM_QUERY_REGEX = /(?:db\.)(?:[a-zA-Z1-9]*)[.]{1}([a-zA-Z1-9]*)/; //extracts the function from a valid query
+const VALID_QUERY_REGEX = new RegExp(/db\.(a-zA-Z1-9_)*.*/);
+const COLLECTION_FROM_QUERY_REGEX = /(?:db\.)([a-zA-Z1-9_]*)[.]{0,1}/; //extracts the collection name from a valid query
+const FUNCTION_FROM_QUERY_REGEX = /(?:db\.)(?:[a-zA-Z1-9_]*)[.]{1}([a-zA-Z1-9]*)/; //extracts the function from a valid query
 
-const EXTRACT_FULL_QUERY_REGEX = /^(?:db).*.(?:[a-zA-Z0-9])\(([^]+.*)\)/; // https://regex101.com/r/sQ9cP4/2
+const EXTRACT_FULL_QUERY_REGEX = /^(?:db)\..*\..*?\((.*)\)/; // https://regex101.com/r/sQ9cP4/2
```

Comment:
:+1:  good call on replace the alphanumeric portion that matches on collection name. Was just checking out this : https://docs.mongodb.org/manual/reference/limits/#Restriction-on-Collection-Names